frontend/src/metabase/hoc/Route.tsx

```diff
@@ -1,44 +0,0 @@
-import { cloneElement } from "react";
-import { Route as _Route } from "react-router";
-import _ from "underscore";
-
-import { useRegisterMetabotContextProvider } from "metabase/metabot";
-
-import title from "./Title";
-
-function metabot(ComposedComponent: React.ComponentType<any>) {
-  return function MetabotRouteContext(props: any) {
-    const { name, description } = props.route;
-
-    useRegisterMetabotContextProvider(() => {
-      if (!name) {
-        return;
-      }
-      return {
-        current_page: {
-          name,
-          ...(description ? { description } : {}),
-        },
-      };
-    }, [name, description]);
-
-    return <ComposedComponent {...props} />;
-  };
-}
-
-// react-router Route wrapper that adds a `title` property
-export class Route extends _Route {
```

Comment:
I got a few grey hairs from this one. The TL:DR; is I don't completely understand why it breaks, but using a HoC of `title` or `metabot` along with `cloneElement` on pages that use `QueryBuilder` it will break routing.

You can repro this with a much simple test. First, run your e2e suite locally for something like `models-revision-history.cy.spec.js`. If you run it now, you should see the test pass. However if you go to `frontend/src/metabase/routes.jsx` and update the route for `/model/:slug` like so:
before
```tsx
<Route path=":slug" component={QueryBuilder} />
```
after:
```tsx
<Route path=":slug" component={QueryBuilder} title="Sadness" />
```
then you will see a failure when you re-run the test.


I was able to repro this with a very simple example. So this Route definition is fine:
```tsx
export class Route extends _Route {
  static createRouteFromReactElement(element: any) {
    const component = element.props.component || (({ children }) => children)
    return (_Route as any).createRouteFromReactElement(
      cloneElement(element, { component }),
    );
  }
}
```
But this one is not:
```tsx
export class Route extends _Route {
  static createRouteFromReactElement(element: any) {
    const hoc = (ComposedComponent) => (props) => <ComposedComponent {...props} />;
    const component = element.props.component || (({ children }) => children)
    return (_Route as any).createRouteFromReactElement(
      cloneElement(element, { component: hoc(component) }),
    );
  }
}
```
Whatever is going on break my understanding of react, and likely doesn't provide much value to keep digging for the time being.


We can get around this by just using `useRegisterMetabotContextProvider` directly to set the values for the time being.